### DISCUSSION TITLE
[Help NEEDED] Not Hooking constructor.

### DISCUSSION THREAD

**Discussion by NULL0B** [LSPosed/LSPosed#197] [Q&A]
Hi Im using Android 11 One UI 3.0. Magisk 21.4 Riru v23.6 and LSposed-v1.0.1 5334-release
Im hooking a class that uses a Singleton Instancer. But result that i never get the constructor hooked. It has to be called at least 1 time. NO? I added Framework(containing class) And Samsung Luncher to Scope (using this feature)
![Anotación 2021-02-24 234111](https://user-images.githubusercontent.com/26231330/109104396-e40e2a80-76f9-11eb-8081-d867c08f061f.png)
Here is the instancer
![Anotación 2021-02-24 234250](https://user-images.githubusercontent.com/26231330/109104476-086a0700-76fa-11eb-825b-c7157f5c5668.png)

**Comment by yujincheng08** [comment]
Are u sure the single instance has not been created when u hook?

**Comment by NULL0B** [comment]
> Are u sure the single instance has not been created when u hook?
Thanks
The instance is already created. But if im hooking all constructors why the hook is not running when the instance was not created? At some time it has to happen.

**Comment by yujincheng08** [comment, marked as answer]
The constructor will only be run once. If it has been run before you hook it, how comes your hooker be called?

**Comment by NULL0B** [comment]
> The constructor will only be run once. If it has been run before you hook it, how comes your hooker be called?

Thats right but how can i hook before to avoid this?

**Reply by yujincheng08** [reply]
try to put it into InitZygote

if it does not work, you have to hook `getInstance` and do what you want and unhook it.

**Reply by NULL0B** [reply]
Thanks
thats what im doing but why the constructor is not hooked? Is LsPosed  hooking too late?

**Reply by yujincheng08** [reply]
InitZygote is a bit earlier. Yet it still happen after the zygote fork.

BTW, all xposed frameworks ensure the hook timing of `handleLoadPackage` but none of them ensures the timing of `initZygote`.

**Reply by NULL0B** [reply]
i tried it on initZygote but had the same result.

**Reply by yujincheng08** [reply]
> if it does not work, you have to hook getInstance and do what you want and unhook it.

**Reply by NULL0B** [reply]
i already know that. But please is it possible make the hook earlier?

**Reply by yujincheng08** [reply]
No. Even Riru cannot ensure its hooking time.
